### PR TITLE
(5.7) PXC-3848: Cluster node crashes when using

### PR DESCRIPTION
**Pull request by kamil-holubicki**
ALTER USER CURRENT_USER()

https://jira.percona.com/browse/PXC-3848

Problem:
1. PXC does not allow using CURRENT_USER() function when operating in
cluster mode.
2. If AuditLog plugin is installed, the error is being logged.
3. As the query contains plain text password, this causes query being
rewritten.
4. Rewrite logic calls get_current_user() function which reports
the error and we end up in point 1 which causes recurrent calls.

Solution:
Allow normal behavior of get_current_user() when the function is called
for query rewrite purpose.

### CODE REVIEW

**Review by venkatesh-prasad-v — APPROVED** [review]

**Comment by dutow** [comment]
`mysql_rewrite_grant` (GRANT .. TO user IDENTIFIED BY) in the same file doesn't have the same issue?

**Comment by kamil-holubicki** [comment]
You are right @dutow. ```GRANT  ... TO CURRENT_USER IDENTIFIED BY ...``` was affected as well. Fixed and updated the testcase.

File: sql/sql_rewrite.cc [hunk 1]
```diff
@@ -385,7 +385,7 @@ void mysql_rewrite_create_alter_user(THD *thd, String *rlb,
 
   while ((tmp_user_name= user_list++))
   {
-    if ((user_name= get_current_user(thd, tmp_user_name)))
+    if ((user_name= get_current_user(thd, tmp_user_name, true)))
```
Comment (dutow) [inline sql/sql_rewrite.cc:397]: Looks good, just one more thing I noticed now: these changes should be in ifdef wsrep blocks.

Comment (kamil-holubicki) [reply sql/sql_rewrite.cc:397]: Yup, exactly the same thing I noticed in the meantime :) Fixed.

**Comment by kamil-holubicki** [comment]
https://pxc.cd.percona.com/job/pxc-5.7-param/280/

**Review by dutow — APPROVED** [review]

**Pull request merged**